OrbitGl/GraphTrack.cpp

```diff
@@ -72,7 +74,27 @@ void GraphTrack::Draw(GlCanvas* canvas, PickingMode /*picking_mode*/) {
   }
 }
 
-//-----------------------------------------------------------------------------
+void GraphTrack::AddValue(double value, uint64_t time) {
+  values_[time] = value;
+  max_ = std::max(max_, value);
+  min_ = std::min(min_, value);
+  value_range_ = max_ - min_;
+
+  if (value_range_ > 0) inv_value_range_ = 1.0 / value_range_;
+}
+
+double GraphTrack::GetValueAtTime(uint64_t time, double default_value) const {
+  auto iterator_lower = values_.lower_bound(time);
+  if (iterator_lower != values_.end()) {
+    return iterator_lower->second;
+  }
+  auto iterator_upper = values_.upper_bound(time);
+  if (iterator_upper != values_.end()) {
```

Comment:
Doesn't `iterator_lower == values_.end()` imply `iterator_upper == values_.end()`? I.e., isn't this always false?

Reply:
yep - it looks to me too like upper_bound here is always pointing to the end()

Reply:
Has this been addressed?